src/components/ScreenWrapper.tsx

```diff
@@ -0,0 +1,252 @@
+import KeyboardAvoidingView from './KeyboardAvoidingView';
+import OfflineIndicator from './OfflineIndicator';
+import SafeAreaConsumer from './SafeAreaConsumer';
+import TestToolsModal from './TestToolsModal';
+
+type ChildrenProps = {
+    insets?: EdgeInsets;
+    safeAreaPaddingBottomStyle?: {
+        paddingBottom?: DimensionValue;
+    };
+    didScreenTransitionEnd: boolean;
+};
+
+type ScreenWrapperProps = {
+    /** Returns a function as a child to pass insets to or a node to render without insets */
+    children: ReactNode | React.FC<ChildrenProps>;
+
+    /** A unique ID to find the screen wrapper in tests */
+    testID: string;
+
+    /** Additional styles to add */
+    style?: StyleProp<ViewStyle>;
+
+    /** Additional styles for header gap */
+    headerGapStyles?: StyleProp<ViewStyle>;
+
+    /** Styles for the offline indicator */
+    offlineIndicatorStyle?: StyleProp<ViewStyle>;
+
+    /** Whether to include padding bottom */
+    includeSafeAreaPaddingBottom?: boolean;
+
+    /** Whether to include padding top */
+    includePaddingTop?: boolean;
+
+    /** Called when navigated Screen's transition is finished. It does not fire when user exit the page. */
+    onEntryTransitionEnd?: () => void;
+
+    /** The behavior to pass to the KeyboardAvoidingView, requires some trial and error depending on the layout/devices used.
+     *  Search 'switch(behavior)' in ./node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js for more context */
+    keyboardAvoidingViewBehavior?: 'padding' | 'height' | 'position';
+
+    /** Whether KeyboardAvoidingView should be enabled. Use false for screens where this functionality is not necessary */
+    shouldEnableKeyboardAvoidingView?: boolean;
+
+    /** Whether picker modal avoiding should be enabled. Should be enabled when there's a picker at the bottom of a
+     *  scrollable form, gives a subtly better UX if disabled on non-scrollable screens with a submit button */
+    shouldEnablePickerAvoiding?: boolean;
+
+    /** Whether to dismiss keyboard before leaving a screen */
+    shouldDismissKeyboardBeforeClose?: boolean;
+
+    /** Whether to use the maxHeight (true) or use the 100% of the height (false) */
+    shouldEnableMaxHeight?: boolean;
+
+    /** Whether to use the minHeight. Use true for screens where the window height are changing because of Virtual Keyboard */
+    shouldEnableMinHeight?: boolean;
+
+    /** Whether to show offline indicator */
+    shouldShowOfflineIndicator?: boolean;
+
+    /**
+     * The navigation prop is passed by the navigator. It is used to trigger the onEntryTransitionEnd callback
+     * when the screen transition ends.
+     *
+     * This is required because transitionEnd event doesn't trigger in the testing environment.
+     */
+    navigation?: StackNavigationProp<RootStackParamList>;
+};
+
+function ScreenWrapper(
+    {
+        shouldEnableMaxHeight = false,
+        shouldEnableMinHeight,
```

Comment:
Should the `shouldEnableMinHeight` also have a default value?
Or if not, why the `shouldEnableMaxHeight` has a default value?

IMO if both aren't required, both should use the same approach.

Reply:
@VickyStash @kubabutkiewicz Before there was no default value for it, it should be safe to add a false default value, wdyt?

Reply:
@blazejkustra Yeah, I think we can add default `false` value